db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java

```diff
@@ -2787,6 +2788,65 @@ public StringDataValue upper(StringDataValue result)
         return result;
     }
 
+    public StringDataValue upperWithLocale(StringDataValue leftOperand, StringDataValue rightOperand,
+                                                StringDataValue result)
+            throws StandardException
+    {
+        return upperLowerWithLocale(leftOperand, rightOperand, result, true);
+    }
+
+    public StringDataValue lowerWithLocale(StringDataValue leftOperand, StringDataValue rightOperand,
+                                                StringDataValue result)
+            throws StandardException
+    {
+        return upperLowerWithLocale(leftOperand, rightOperand, result, false);
+    }
+
+    private StringDataValue upperLowerWithLocale(StringDataValue leftOperand, StringDataValue rightOperand,
+                                           StringDataValue result, boolean isUpper)
+                            throws StandardException
+    {
+        if (result == null)
+        {
+            result = (StringDataValue) getNewNull();
+        }
+
+        if (this.isNull())
+        {
+            result.setToNull();
+            return result;
+        }
+
+        if (leftOperand == null || leftOperand.isNull() || leftOperand.getString() == null)
+        {
+            result.setToNull();
+            return result;
+        }
+
+        if (rightOperand == null || rightOperand.isNull() || rightOperand.getString() == null) {
+            throw StandardException.newException(
+                    SQLState.LANG_INVALID_FUNCTION_ARGUMENT, "NULL", isUpper?"UPPER":"LOWER");
+        }
+        String localeStr = rightOperand.getString();
+        Locale locale = Locale.forLanguageTag(localeStr);
+        if (!validLocale(locale)) {
+            throw StandardException.newException(
+                    SQLState.LANG_INVALID_FUNCTION_ARGUMENT, localeStr, isUpper?"UPPER":"LOWER");
```

Comment:
splice> values UPPER('abc', 'a');
ERROR 22008: 'a' is an invalid argument to the UPPER function.

.... would it be useful to give more information here, like "'a' is not a valid locale." ?

Reply:
Exception with SQLState.LANG_INVALID_FUNCTION_ARGUMENT give error information like this. I think it is better to not add a new SQLState.